File: Codewars/Python/7 kyu/II/Method For Counting Total Occurence Of Specific Digits.py
class List(object):
    @staticmethod
    def count_spec_digits(integers_list, digits_list):
        return [(n, integers_list.count(n)) for n in digits_list if integers_list]




integers_list =  [1, 1, 2, 3 , 1 , 2 , 3 , 4]
digits_list = [1, 3]
integers_list2 = [-18, -31, 81, -19, 111, -888]
digits_list2 = [1, 8, 4]
a = List.count_spec_digits(integers_list, digits_list)
b = List.count_spec_digits(integers_list2, digits_list2)

print(a)
print(b)

File: Codewars/Javascript/7 kyu/II/Method For Counting Total Occurence Of Specific Digits.js
function List() {
  this.countSpecDigits = function (integersList, digitsList) {
    return digitsList.map((d) => {
      let occurrence = integersList.filter(n => `${n}`.includes(d));
      let total = occurrence.map(c => `${c}`.split('').filter(n => n == d).join('')).join('').length
      return [d, total]
    })
  };
}



integersList = [1, 1, 2, 3, 1, 2, 3, 4];
digitsList = [1, 3];


integersList2 = [-18, -31, 81, -19, 111, -888];
digitsList2 = [1, 8, 4];

let l = new List()
let l2 = new List()

// console.log(l.countSpecDigits(integersList, digitsList))
console.log(l2.countSpecDigits(integersList2, digitsList2))
